src/modelgauge/sut_factory.py

```diff
@@ -0,0 +1,98 @@
+from enum import Enum
+
+from modelgauge.config import load_secrets_from_config
+from modelgauge.dynamic_sut_factory import DynamicSUTFactory, UnknownSUTMakerError
+from modelgauge.dynamic_sut_metadata import DynamicSUTMetadata
+from modelgauge.secret_values import RawSecrets
+from modelgauge.sut import SUT
+from modelgauge.sut_registry import SUTS
+from modelgauge.suts.huggingface_sut_factory import HuggingFaceSUTFactory
+from modelgauge.suts.openai_sut_factory import OpenAISUTFactory
+from modelgauge.suts.together_sut_factory import TogetherSUTFactory
+
+
+class SUTNotFoundException(Exception):
+    pass
+
+
+class SUTType(Enum):
+    DYNAMIC = "dynamic"
+    KNOWN = "known"
+    UNKNOWN = "unknown"
+
+
+# TODO: Auto-collect. Maybe make "driver" keys a constant in each factory module.
```

Comment:
They are, at least in some. E.g. huggingface_sut_factory line 19. That was the intent originally.